spring-cloud-gcp-data-firestore/src/test/java/com/google/cloud/spring/data/firestore/SimpleFirestoreReactiveRepositoryTests.java

```diff
@@ -16,21 +16,32 @@
 
 package com.google.cloud.spring.data.firestore;
 
-import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
 
-import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
 
 public class SimpleFirestoreReactiveRepositoryTests {
 
 	@Test
-	public void deleteAllByIdUnimplemented() {
+	public void deleteAllById() {
 		FirestoreTemplate mockTemplate = mock(FirestoreTemplate.class);
-		SimpleFirestoreReactiveRepository<String> repository =
-				new SimpleFirestoreReactiveRepository<>(mockTemplate, String.class);
-		assertThatThrownBy(() -> repository.deleteAllById(new ArrayList<>()))
-				.isInstanceOf(UnsupportedOperationException.class);
+		SimpleFirestoreReactiveRepository<String> repository = new SimpleFirestoreReactiveRepository<>(mockTemplate,
+				String.class);
+		Iterable<String> idList = Arrays.asList("1", "2");
+		repository.deleteAllById(idList);
```

Comment:
I'd leave a comment above this line saying that the test is only testing that the request was passed through to `FirestoreTemplate`, so the returned `Publisher` does not need to be subscribed to. It's a bit jarring to see a publisher output discarded -- it would lead to the functionality never being invoked in real life.

Alternatively, you could do what you've done in `FirestoreTemplateTests` -- set up a when/then mock structure for `mockTemplate` that *only* matches the correct publisher input, and then instead of doing argument captor below, a `StepVerifier` can be used to subscribe and validate that the response of `repository.deleteAllById` matches the output that mockito would only give on correct parameters.

Reply:
Tried with this alternative route, but stuck on comparing two publisher input: `Flux` equals is inherited from Object, I cannot find easy ways to compare the input for mock. The argument captor looks more intuitive to me.
Going to just add the comment instead.